hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -241,38 +244,24 @@ async def async_map(self,
             fn = chunk(fn)
             iterables = iterables_chunks
 
-        tasks = [asyncio.create_task(self.async_submit(fn, *arguments))
-                 for arguments in zip(*iterables)]
-
+        submit_tasks = [asyncio.ensure_future(self.async_submit(fn, *arguments))
+                        for arguments in zip(*iterables)]
         try:
-            bp_futures: List[BatchPoolFuture] = await asyncio.gather(*tasks)
-        finally:
-            _, exc, _ = sys.exc_info()
-            if exc is not None:
-                for task in tasks:
-                    if not task.done():
-                        task.cancel()
-
-                if tasks:
-                    await asyncio.wait(tasks)
-
-                bp_futures = []
-                for task in tasks:
-                    assert task.done()
-                    if not task.cancelled() and not task.exception():
-                        bp_future = task.result()
-                        bp_futures.append(bp_future)
-
-                if bp_futures:
-                    await asyncio.gather(*[bp_fut.async_cancel() for bp_fut in bp_futures], return_exceptions=True)
+            bp_futures = [await t for t in submit_tasks]
```

Comment:
This is not in parallel anymore. Was it intentional to do this?

Reply:
The `ensure_future` on line 247 places all the tasks into the background. Each await here suspends execution and lets one of those background tasks make progress.